src/libcollections/bit.rs

```diff
@@ -604,6 +605,115 @@ impl BitVec {
         Iter { bit_vec: self, next_idx: 0, end_idx: self.nbits }
     }
 
+    /// Moves all bits from `other` into `Self`, leaving `other` empty.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// # #![feature(collections)]
+    /// use std::collections::BitVec;
+    ///
+    /// let mut a = BitVec::new();
+    /// a.push(true);
+    /// a.push(false);
+    ///
+    /// let mut b = BitVec::new();
+    /// b.push(false);
+    /// b.push(true);
+    /// b.push(true);
+    ///
+    /// a.append(&mut b);
+    ///
+    /// assert_eq!(a.len(), 5);
+    /// assert_eq!(b.len(), 0);
+    /// assert!(a.eq_vec(&[true, false, false, true, true]));
+    /// ```
+    #[unstable(feature = "collections",
+               reason = "recently added as part of collections reform 2")]
+    pub fn append(&mut self, other: &mut Self) {
+        let b = self.len() % u32::BITS;
+
+        self.nbits += other.len();
+        other.nbits = 0;
+
+        if b == 0 {
+            self.storage.append(&mut other.storage);
+        } else {
+            self.storage.reserve(other.storage.len());
+
+            for block in other.storage.drain() {
+                *(self.storage.last_mut().unwrap()) += block << b;
```

Comment:
This threw me off for a bit. `&`ing makes more semantic sense to me, unless you have a strong argument otherwise.